web/db.py

```diff
@@ -1198,10 +1198,18 @@ def _process_insert_query(self, query, tablename, seqname):
                 seqname = None
 
         if seqname:
-            query += "; SELECT currval('%s')" % seqname
+            query += self.get_sequence_query(seqname)
 
         return query
 
+    def get_sequence_query(self, seqname):
+        import re
+        # Ensure the sequence name is valid
+        if not re.match(r'^[a-zA-Z_][a-zA-Z0-9_]*$', seqname):
```

Comment:
```suggestion
        if not re.match(r'^[a-zA-Z_][a-zA-Z0-9_$]*$', seqname):
```
I am just basing this suggestion on PosgreSQL documentation section [4.4.1 Identifiers and Key Words](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS), which says in relevant part:
> SQL identifiers and key words must begin with a letter (a-z, but also letters with diacritical marks and non-Latin letters) or an underscore (_). Subsequent characters in an identifier or key word can be letters, underscores, digits (0-9), or dollar signs ($). Note that dollar signs are not allowed in identifiers according to the letter of the SQL standard, so their use might render applications less portable. The SQL standard will not define a key word that contains digits or starts or ends with an underscore, so identifiers of this form are safe against possible conflict with future extensions of the standard.

However, I take the point that the SQL standard does not allow `$`, so maybe it's not worth including?